### DISCUSSION TITLE
{% exit %} to dump / expected way to dump in templates

### DISCUSSION THREAD

**Discussion by Anubarak** [craftcms/cms#13848] [General]
Just wanted to know what your expected way to dump in templates is.
Before Craft 4.0.1 I always used the `{% exit %}` token and had this one in my templates

```twig
<pre>
{{ dump("test") }}
</pre>  
{% exit %}
```

This doesn't work anymore due to this change
https://github.com/craftcms/cms/blob/4.0.1/src/web/Application.php#L721C8-L735

So my question is: was it intended to remove the ability to debug / output things via `{% exit %}` because there is nothing in the documentation https://craftcms.com/docs/4.x/dev/tags.html#exit it just states that exit will end the request, not that it will prevent any output.

I liked to use `exit` because it gave me the possibility to see what was rendered until this point. `{% dd %}` kills the previous output and one of the first things I always do in Craft is to remove those lines so I can use `exit` again in my dev environment.
Maybe you could pass an optional parameter in your `end` function to NOT throw an ExitException or add an additional `endTemplate` function that just calls Yii2 `end` function?

**Comment by brandonkelly** [comment]
Thanks for pointing that out! That was an unintentional change in behavior in Craft 4. Fixed now for the next release (1fde3254c9fb378c3076acfc0b953c9d1f454424).

The other dumping tool that might be helpful is the new [`{% dump %}`](https://craftcms.com/docs/4.x/dev/tags.html#dump) tag, which will dump the variable out into the debug toolbar, rather than directly on the page.

**Reply by brandonkelly** [reply]
Craft 4.5.8 is out with that fix 🎉

**Reply by Anubarak** [reply]
Awesome as always.
Thank you very much 👍
I really appreciate that
